files/spark/bin/charmed-spark-entrypoint.sh

```diff
@@ -1,9 +1,24 @@
 #!/bin/bash
 
-# This script still exists because passing of working directory to services has not yet landed
-# in Pebble (https://github.com/canonical/pebble/issues/158) and being in the write directory
-# is important for the entrypoint script
+TYPE=$1
 
-cd /opt/spark
+echo "Running script with ${TYPE} flavour"
 
-./entrypoint.sh $*
\ No newline at end of file
+case "${TYPE}" in
+  driver|executor)
+    cd /opt/spark
+    ./entrypoint.sh $*
+    ;;
```

Comment:
Maybe use `pushd` instead of `cd` here

Reply:
Also, `$*` needs to be quoted, `"$*"` or use `$@`

Reply:
+1 - `"$@"` is probably the safest route.